frontend/src/lib/components/SelectBox.tsx

```diff
@@ -0,0 +1,248 @@
+            }
+            if (e.key === 'Enter') {
+                actions.clickSelectedItem(values.selectedItem)
+            }
+        },
+    }),
+})
+
+export function SelectBox({
+    items,
+    selectedItemKey,
+    onSelect,
+    onDismiss,
+}: {
+    items: SelectBoxItem[]
+    selectedItemKey: string
+    onSelect: CallableFunction
+    onDismiss: CallableFunction
+}): JSX.Element {
+    const dropdownRef = useRef()
+    const dropdownLogic = actionFilterDropdownLogic({ updateFilter: onSelect, items })
+    const { selectedItem, RenderInfo } = useValues(dropdownLogic)
+    const { setSearch, setSelectedItem, onKeyDown } = useActions(dropdownLogic)
+
+    const deselect = (e): void => {
+        if (dropdownRef.current.contains(e.target)) {
+            return
+        }
+        onDismiss && onDismiss(e)
+    }
+
+    useEffect(() => {
+        if (selectedItemKey) {
+            const allSources = items.map((item) => item.dataSource).flat()
+            setSelectedItem(allSources.filter((item) => item.key === selectedItemKey)[0] || false)
+            const offset = document.querySelector('.action-filter-dropdown [datakey="' + selectedItemKey + '"]')
+                ?.offsetTop
+            document.querySelector('.search-list').scrollTop = offset
+        }
+        document.addEventListener('mousedown', deselect)
+        document.addEventListener('keydown', onKeyDown)
+        return () => {
+            document.removeEventListener('mousedown', deselect)
+            document.removeEventListener('keydown', onKeyDown)
+        }
+    }, [])
+    return (
+        <div ref={dropdownRef} className="action-filter-dropdown">
+            <Row style={{ height: '100%' }}>
+                <Col sm={14} style={{ borderRight: '1px solid rgba(0, 0, 0, 0.1)', maxHeight: '100%' }}>
+                    <Input
+                        placeholder="Search events"
+                        autoFocus
+                        onChange={(e) => {
+                            setSearch(e.target.value)
+                        }}
+                        style={{ width: '100%', borderRadius: 0 }}
+                    />
+                    <div className="search-list">
+                        {items.map((item) => (
+                            <>
+                                <SelectUnit
+                                    name={item.name}
+                                    dropdownLogic={dropdownLogic}
+                                    dataSource={item.dataSource}
+                                />
+                                <Divider />
+                            </>
+                        ))}
+                    </div>
+                </Col>
+                <Col sm={10} className="info-box">
+                    {RenderInfo && <RenderInfo item={selectedItem} />}
+                </Col>
+            </Row>
+        </div>
+    )
+}
+
+export function SelectUnit({
+    name,
+    dataSource,
+    dropdownLogic,
```

Comment:
Q: Why is this a prop? Could find no other instances where we pass logics around this way besides this PR but I didn't look super hard.

Reply:
Hm take a look at PropertyFilter.js for example. It's a way of making sure the props for the logic are passed through correctly everywhere.